apps/dashboard/src/components/activity/activity-detail-card.tsx

```diff
@@ -0,0 +1,44 @@
+import { cn } from '@/utils/ui';
+import { ChevronDown } from 'lucide-react';
+import { ReactNode, useState } from 'react';
+
+interface ActivityDetailCardProps {
+  title: ReactNode;
+  timestamp?: string;
+  expandable?: boolean;
+  open?: boolean;
+  children?: ReactNode;
+}
+
+export function ActivityDetailCard({ title, timestamp, expandable = false, open, children }: ActivityDetailCardProps) {
+  const [internalOpen, setInternalOpen] = useState(false);
+  const isExpanded = open ?? internalOpen;
```

Comment:
We have the accordion primitive that we can use for the nice animation. However, I don't know if it will require doing some updates.